### PR TITLE
Specify coordinate space for all geometry.

### PR DESCRIPTION
**Pull request by szager-chromium**
Issue #24

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -455,8 +455,9 @@ dictionary IntersectionObserverInit {
 		and vice versa.
 
 		Threshold values must be in the range of [0, 1.0] and represent a
-		percentage of the area as specified by
-		<a>target</a>.{{Element/getBoundingClientRect()}}.
+		percentage of the area of the rectangle produced
+		by running the <a>getBoundingClientRect</a> algorithm
```
Comment (tabatkins) [inline index.bs]: There's no way this autolink works.

File: index.bs [hunk 2]
```diff
@@ -597,10 +595,8 @@ To <dfn>run the update intersection observations steps</dfn> for an
 		2. If the <a>intersection root</a> is not the <a>implicit root</a>,
 			and |target| is not in the same document as the <a>intersection root</a>,
 			skip further processing for |target|.
-		3. Let |targetRect| be a {{DOMRectReadOnly}}
-			obtained by the same algorithm as that of {{Element}}'s
-			{{Element/getBoundingClientRect()}} performed on
-			|target|.
+		3. Let |targetRect| be a {{DOMRectReadOnly}} obtained
+			by running the <a>getBoundingClientRect</a> algorithm on |target|.
```
Comment (tabatkins) [inline index.bs]: Same - no way this works. (If it does, something's fucked up.)

**Comment by ojanvafai** [comment]
lgtm once you address tabs comments.

File: index.bs (hunk 1 above)
Comment (szager-chromium) [inline index.bs]: Look at the html, bruh.

**Comment by szager-chromium** [comment]
Tab, can you expand on that?  There were already references to getBoundingClientRect before this change.  It's defined as:

urlPrefix: http://dev.w3.org/csswg/cssom-view/
	url: #dom-element-getboundingclientrect; type:dfn; text: getBoundingClientRect

File: index.bs (hunk 1 above)
Comment (tabatkins) [inline index.bs]: Ugh, line 52 in the anchors block is wrong/confusing. It should be `type: method; text: getBoundingClientRect(); for: Element`.  What it currently says is nonsense.

**Comment by szager-chromium** [comment]
Fixed, PTAL

**Comment by tabatkins** [comment]
gBCR() is a method of Element, not a "dfn" by itself. It's just marked up incorrectly in the anchors block, allowing this confusing markup.

It's not even needed - CSSOM is already in the linking database.  We need to drop it from the anchors block and just mark up the autolinks correctly.

Same with the "pinch zoom" definition, fwiw - it's not exported (which is a bug in the spec), but it can be used more directly with a link-defaults block:

```
<pre class=link-defaults>
spec:cssom-view-1; type:dfn; text:pinch zoom
</pre>
```

**Comment by szager-chromium** [comment]
I am flying blind with this bikeshed stuff, I really have no idea what I'm doing.  If you could just tell me what to do, that would be great.

**Comment by szager-chromium** [comment]
I'm getting this from bikeshed:

WARNING: Multiple possible 'functionish' refs for 'getBoundingClientRect()'.
Arbitrarily chose the one in cssom-view-1.
If this is wrong, insert one of the following lines into a <pre class=link-defaults> block:
spec:cssom-view-1; type:method; for:Element; text:getBoundingClientRect()
spec:cssom-view-1; type:method; for:Range; text:getBoundingClientRect()


So I added this:

<pre class="link-defaults">
spec: cssom-view-1; type: method; for: Element; text: getBoundingClientRect()
</pre>

Now I get this:

FATAL ERROR: No 'method' refs found for 'getBoundingClientRect()' with for='[u'Element']'.

So, basically, I'm lost.

**Comment by szager-chromium** [comment]
html comments... sigh

**Comment by szager-chromium** [comment]
WARNING: Multiple possible 'functionish' refs for 'getBoundingClientRect()'.
Arbitrarily chose the one in cssom-view-1.
If this is wrong, insert one of the following lines into a &lt;pre class=link-defaults&gt; block:
spec:cssom-view-1; type:method; for:Element; text:getBoundingClientRect()
spec:cssom-view-1; type:method; for:Range; text:getBoundingClientRect()

So I added:

&lt;pre class="link-defaults"&gt;
spec: cssom-view-1; type: method; for: Element; text: getBoundingClientRect()
&lt;/pre&gt;

Now I get:

FATAL ERROR: No 'method' refs found for 'getBoundingClientRect()' with for='[u'Element']'.

**Comment by tabatkins** [comment]
It sounds like your Bikeshed is out of date?

**Comment by szager-chromium** [comment]
PTAL

**Comment by tabatkins** [comment]
Here's what needs to happen, which works in a recent Bikeshed:

1. Remove the cssom-view things from the anchors block.
2. Add a link-defaults block:
    ```
    <pre class="link-defaults">
    spec: cssom-view-1; type: dfn; text: pinch zoom
    </pre>
    ```
3. Make sure you're linking with the right autolink: `{{Element/getBoundingClientRect()}}`. (The file previously used this in several places, you changed them to an `<a>` in this PR.)

File: index.bs [hunk 3]
```diff
@@ -60,6 +57,8 @@ spec:dom-ls
 spec: css-values-3; type: dfn
 	text: absolute length
 	text: dimension
+spec: cssom-view-1; type: dfn; text: pinch zoom
+spec: cssom-view-1; type: method; for: Element; text: getBoundingClientRect()
```
Comment (tabatkins) [inline index.bs]: Kill this and just specify that it's for Element at each call site.

File: index.bs [hunk 4]
```diff
@@ -381,31 +380,31 @@ dictionary IntersectionObserverEntryInit {
 <div dfn-type="attribute" dfn-for="IntersectionObserverEntry">
 	: <dfn>boundingClientRect</dfn>
 	::
-		A {{DOMRectReadOnly}} obtained by the same algorithm
-		as that of {{Element}}'s {{Element/getBoundingClientRect()}}
-		performed on {{IntersectionObserverEntry/target}}.
+		A {{DOMRectReadOnly}} obtained by running the <a>getBoundingClientRect()</a> algorithm
```
Comment (tabatkins) [inline index.bs]: Use {{Element/getBoundingClientRect()}}.

**Comment by szager-chromium** [comment]
Done, PTAL

File: index.bs [hunk 5]
```diff
@@ -60,6 +57,9 @@ spec:dom-ls
 spec: css-values-3; type: dfn
 	text: absolute length
 	text: dimension
+spec: cssom-view-1
+	type: dfn; text: pinch zoom
+	type: method; for: Element; text: getBoundingClientRect()
```
Comment (tabatkins) [inline index.bs]: Kill this line, it's not needed.

**Comment by tabatkins** [comment]
Final nit, then lgtm. ^_^

**Comment by szager-chromium** [comment]
PTAL

**Pull request merged**